DOCS-2623: Document payment button ownership (#1675)
MultiSafepay/docs: content/payments/checkout/payment-components/multiple/multiple-3.md

read_more: '.'
--- 

## Handle the interaction

{{< blue-notice >}} This step only applies if using your own or an existing payment button. {{< /blue-notice >}}

**1.** Assign the button element to a variable:

```
const paymentButton = document.querySelector('#payment-button');
```

**2.** Create an event handler for the payment button:

- When the customer clicks the payment button, call the `getPaymentData()` method.

> - Send the response to the call to your server (`createOrder`) to create an order (see [Create an order](#create-an-order)).

Author: ```suggestion
- Send the response to your server and create an order. See [Create an order](#create-an-order)).
```

- Return the reponse from your server to the client-side to redirect the customer.

```
paymentButton.addEventListener('click', e => {
    paymentButton.addAttribute('disabled');
    if (PaymentComponent.hasErrors()) {
        let errors = PaymentComponent.getErrors();
        console.log(errors);
        return false;
    }
    createOrder(PaymentComponent.getPaymentData()).then(response => {
        if(!response || !response.success) {
            paymentButton.removeAttribute('disabled');
            console.log(response);
        } else {
            PaymentComponent.init('redirection', {
                order: response.data
            });
        }
    });
});
```

**Note:** When using your own payment button, if the customer clicks it again during the latency before redirection, this creates duplicate orders. 

To avoid duplicate orders, disable the button until you have attempted to create an order. Then, check `response.success`:

- If `true`, don't re-enable the button and proceed to the redirect.
- If `false`, re-enable the button for the customer to try again. 

{{< details title="Redirect to 3D verification" >}}

> The `init('redirection')` method redirects customers that pay with credit card to the relevant page.

Author: ```suggestion
The `init('redirection')` method redirects customers who pay by credit card to the relevant page.
```
